src/builder/QuarkBuilder.sol

```diff
@@ -345,4 +374,25 @@ contract QuarkBuilder {
             }
         }
     }
+
+    function totalAvailableAsset(
```

Comment:
Sounds good. 

I think I left it here because I see `assertSuffientFunds` and `assertFundsAvailable`  Make me wonder if we should move all that to Accounts.sol as well.  🤔  (or strip off some account-only logics into Accounts.sol)